compiler/semantic/expr.go

```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/brimdata/zed/zson"
 )
 
-func semExpr(scope *Scope, e ast.Expr) (dag.Expr, error) {
+func (c *checker) semExpr(scope *Scope, e ast.Expr) (dag.Expr, error) {
```

Comment:
Any reason not to drop the scope parameter everywhere in favor of checker.scope?

Reply:
Yeah I was just thinking the same thing, I'll do this.